lib/multi_segment_linear_indicator.dart

```diff
@@ -0,0 +1,276 @@
+import 'package:flutter/material.dart';
+
+/// A multi-segment linear progress indicator that displays three segments with different colors and styles.
+/// One segment can optionally display stripes.
+///
+/// The total of all segments must be less than or equal to 1.0 (100%).
+/// Each segment percentage must be between 0.0 and 1.0.
+///
+/// Example:
+/// ```dart
+/// MultiSegmentLinearIndicator(
+///   segments: [
+///     SegmentLinearIndicator(percent: 0.3, color: Colors.red, enableStripes: true),
+///     SegmentLinearIndicator(percent: 0.4, color: Colors.blue),
+///     SegmentLinearIndicator(percent: 0.3, color: Colors.green),
+///   ],
+///   lineHeight: 20,
+///   barRadius: Radius.circular(10),
+///   animation: true,
+///   enableStripes: [1, 2], // Enable stripes for first and second segments
+/// )
+/// ```
+
+/// Represents a segment in the linear indicator with a percentage and color.
+class SegmentLinearIndicator {
+  final double percent;
```

Comment:
can you add some validations for `percent` value? the sume of them must be equal or lower than 1.0